src/pages/recurring-expenses/common/hooks.tsx

```diff
@@ -0,0 +1,467 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Expense } from 'common/interfaces/expense';
+import { StatusBadge } from 'components/StatusBadge';
+import recurringExpenseStatus from 'common/constants/recurring-expense';
+import recurringExpensesFrequency from 'common/constants/recurring-expense-frequency';
+import { useTranslation } from 'react-i18next';
+import { date, endpoint } from 'common/helpers';
+import { useCurrentCompanyDateFormats } from 'common/hooks/useCurrentCompanyDateFormats';
+import { EntityStatus } from 'components/EntityStatus';
+import { useCurrentCompany } from 'common/hooks/useCurrentCompany';
+import { useFormatMoney } from 'common/hooks/money/useFormatMoney';
+import { Link } from '@invoiceninja/forms';
+import { route } from 'common/helpers/route';
+import { Divider } from 'components/cards/Divider';
+import { DropdownElement } from 'components/dropdown/DropdownElement';
+import { Action } from 'components/ResourceActions';
+import { useAtom } from 'jotai';
+import { useNavigate } from 'react-router-dom';
+import { recurringExpenseAtom } from './atoms';
+import { RecurringExpense } from 'common/interfaces/recurring-expense';
+import { RecurringExpenseStatus } from 'common/enums/recurring-expense-status';
+import { request } from 'common/helpers/request';
+import { toast } from 'common/helpers/toast/toast';
+import { useQueryClient } from 'react-query';
+import { expenseAtom } from 'pages/expenses/common/atoms';
+import paymentType from 'common/constants/payment-type';
+import { customField } from 'components/CustomField';
+import { useCurrentUser } from 'common/hooks/useCurrentUser';
+
+export type DataTableColumnsExtended<TResource = any, TColumn = string> = {
+  column: TColumn;
+  id: keyof TResource;
+  label: string;
+  format?: (field: string | number, resource: TResource) => unknown;
+}[];
+
+export const recurringExpenseColumns = [
+  'status',
+  'number',
+  'vendor',
+  'client',
+  'date',
+  'amount',
+  'public_notes',
+  'entity_state',
+  'archived_at',
+  //   'assigned_to', @Todo: Need to resolve relationship
+  //   'category', @Todo: Need to resolve relationship
+  'created_at',
```

Comment:
I kept this comments same as for expense entity, please let me know if you have any updates here.